witchcraft-server/src/debug/heap_profile.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::debug::Diagnostic;
+use bytes::Bytes;
+use conjure_error::Error;
+use http::HeaderValue;
+use refreshable::Refreshable;
+use std::{
+    ffi::{c_char, CString},
+    fs,
+};
+use tempfile::NamedTempFile;
+use witchcraft_log::info;
+use witchcraft_server_config::runtime::RuntimeConfig;
+
+#[no_mangle]
+#[allow(non_upper_case_globals)]
+static malloc_conf: &c_char = unsafe { &*c"prof:true,prof_active:false".as_ptr() };
```

Comment:
This is a magic static that controls the startup featureset of jemalloc. `prof:true` enables the ability to turn on profiling, and `prof_active:false` means that it's not actually sampling allocations until we turn it on below.